### PR TITLE
[i2c] Do not make timeouts persist.

### PR DESCRIPTION
**Pull request by tjaychen**
After timeout (and interrupt triggered), recount the timeout period again instead of allowing it to immediately re-trigger.

Signed-off-by: Timothy Chen <timothytim@google.com>

### CODE REVIEW

**Comment by jdonjdon** [comment]
I reran #16495 
./util/dvsim/dvsim.py ./hw/ip/i2c/dv/i2c_sim_cfg.hjson -i i2c_target_timeout -r 1 -v m -s 1 -w fsdb
with this update and @271589707ps,  tb.dut.i2c_core.u_i2c_fsm.sda_o go low and never comes back.
is it waiting for scli low?
image

**Comment by tjaychen** [comment]
did you change anything in the test? i was not able to reproduce this.

**Comment by jdonjdon** [comment]
> did you change anything in the test? i was not able to reproduce this.

Hmm.. no I just used #16495 + this rtl change.
maybe I rebased my local branch. Let me push #16495 again. Try now.
you will see following.

UVM_FATAL @ 10272438191 ps: (i2c_target_smoke_vseq.sv:75) [target_smoke_vseq] wait timeout occurred!

**Comment by tjaychen** [comment]
sounds good, let me try again.

**Comment by tjaychen** [comment]
ah okay you're right.
It looks like the host kept the clock high during the the target transmit phase.
The target transmit across the entire high pulse of a `scl`. Since there was never a `negedge`, the target never stops transmitting, that's why we see `sda_o` always 0 here, since it happened to be the value it wanted.

**Review by jdonjdon — APPROVED** [review]
LGTM

**Comment by tjaychen** [comment]
updated with idle fix.
